app/uk/gov/hmrc/excisemovementcontrolsystemapi/controllers/SubmitMessageController.scala

```diff
@@ -103,11 +104,19 @@ class SubmitMessageController @Inject()(
                          (implicit hc: HeaderCarrier): EitherT[Future, Result, EISSubmissionResponse] = {
     workItemService.addWorkItemForErn(authorisedErn, fastMode = true)
 
-    for {
-      result <- EitherT(submissionMessageService.submit(request, authorisedErn))
-      _ <- auditService.auditMessage(request.ieMessage)
-    } yield result
-
+    EitherT {
+      submissionMessageService.submit(request, authorisedErn).map {
+        case Left(result) => auditService.auditMessage(request.ieMessage, "Failed to Submit")
+          Left(result)
+        case Right(response) => auditService.auditMessage(request.ieMessage)
+          Right(response)
+      }.recover {
```

Comment:
do we need this recover? Submit will cache any exception for the message submission and the NRS and return a left. So this recover may be redundant. auditMessage also will cache an exception and return ( HandlerResult.Failure).

Reply:
Not used in the tests currently and can't make it go in there so gonna remove for now.